airstack-modules/modules/airstack/social/social.ts

```diff
@@ -0,0 +1,262 @@
+import { BigInt, ethereum } from '@graphprotocol/graph-ts';
+import {
+  AirExtra,
+  AirUser,
+  AirProfile,
+  AirBlock,
+  AirUserRegisteredTransaction,
+} from '../../../generated/schema';
+import { getOrCreateAirAccount, getOrCreateAirBlock, getChainId, updateAirEntityCounter, getOrCreateAirToken, createAirExtra } from '../common/index';
+import { AirProtocolType, AirProtocolActionType, AIR_USER_REGISTERED_TRANSACTION_ENTITY_COUNTER_ID } from './utils';
+
+export namespace social {
+
+  /**
+   * @dev this function tracks a air user and profile registered transaction
+   * @param block ethereum block
+   * @param transactionHash transaction hash
+   * @param logOrCallIndex log or call index
+   * @param fromAddress air transaction from which user token was transferred
+   * @param toAddress air transaction to which user token was transferred
+   * @param tokenId tokenId of the profile token that was transferred - ERC721
+   * @param tokenAddress token address of the profile token that was transferred - ERC721
+   * @param dappUserId dappUserId (eg: farcasterId)
+   * @param profileName profile name
+   * @param profileExtras extra data (eg: farcaster tokenUri)
+   * @param userExtras extra data (eg: farcaster homeUrl and recoveryAddress)
+   */
+  export function trackUserAndProfileRegisteredTransaction(
+    block: ethereum.Block,
+    transactionHash: string,
+    logOrCallIndex: BigInt,
+    fromAddress: string,
+    toAddress: string,
+    tokenId: string,
+    tokenAddress: string,
+    dappUserId: string,
+    profileName: string,
+    profileExtras: AirExtraData[],
+    userExtras: AirExtraData[],
+  ): void {
+    let chainId = getChainId();
+    // creating air block
+    let airBlock = getOrCreateAirBlock(chainId, block.number, block.hash.toHexString(), block.timestamp);
+    airBlock.save();
+    // creating air user
+    let userId = chainId.concat('-').concat(dappUserId);
+    // create air user extras
+    let airUserExtras = new Array<AirExtra>();
+    for (let i = 0; i < userExtras.length; i++) {
+      let extra = userExtras[i];
+      let extraId = userId.concat("-").concat(extra.name);
+      let airUserExtraData = createAirExtra(
+        extra.name,
+        extra.value,
+        extraId,
+      );
+      airUserExtraData.save();
+      airUserExtras.push(airUserExtraData);
+    }
+    // create air profile extras
+    let airProfileExtras = new Array<AirExtra>();
+    for (let i = 0; i < profileExtras.length; i++) {
+      let extra = profileExtras[i];
+      let extraId = userId.concat("-").concat(extra.name);
+      let airProfileExtraData = createAirExtra(
+        extra.name,
+        extra.value,
+        extraId,
+      );
+      airProfileExtraData.save();
+      airProfileExtras.push(airProfileExtraData);
+    }
+    // get air extra data ids
+    let airUserExtraIds = getAirExtraDataEntityIds(airUserExtras);
```

Comment:
You already have the  for loop you can create the `airUserExtraIds` array in the loop itself.

Reply:
done